Mothercell/Hive.IO/Hive.IO/Plots/OperationalPerformancePlot.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Drawing;
+using Grasshopper.GUI.Canvas;
+using Grasshopper.Kernel;
+using Rhino;
+
+namespace Hive.IO.Plots
+{
+    public delegate double QueryResults(Results results);
```

Comment:
note for your delegate that `Results` might contain other objects (`Hive.IO.EnergySystems`, `Hive.IO.Building`, etc... not final yet, but that's the concept), and not properties of type `double` anymore. So e.g. if you wanna get heating loads, you will need to read something like `Results.Building.Zone[i].HeatingLoads` . 

I can make getters in the `Results` class tho, so the API doesn't become so nested!

Reply:
The delegates still need to be written. using the delegate system, the visualizer is decoupled from the `Results` object, in that the delegate will define _how_ to actually query the results. So have fun! And when you're ready, we can plug it in!